### ISSUE TITLE
[llvm] Depends on graphviz

### ISSUE THREAD

**Issue by blu** [skycocker/chromebrew#2711]
Guys, latest llvm package (7.0.0) depends on package graphviz, which, in its turn depends on libx11 and a whole bunch of packages that should not really have anything to do with llvm (that includes mesa!)

What mandates that dependency?

## Versions and system information
chromebrew: 1.0.3

**Comment by JL2210** [comment]
Wow! I have no idea.

**Comment by uberhacker** [comment]
@blu: The graphviz dependency is for lldb which is now integrated in with llvm.  I guess this is needed for graphical debugging which makes sense to enhance the visualization.

**Comment by blu** [comment]
Ok, but that causes a bit of an issue. Aside form the overgrowing of the llvm installation (llvm by itself is not small, but now it brings along x11, mesa and bunch of extra GUI stuff), mesa breaks the native /usr/lib/libGLESv2, and the latter is required for the functioning of non-intel GPUs. So now the first thing that needs to be done on a non-x86 chromebook after installing llvm is remove mesa.

Do we need to provide GUI frontends/utils to lldb by default? It's not like lldb does not work in a terminal. To make a parallel, we don't bring in any default IDEs just because somebody asked for gcc or llvm : )

**Comment by uberhacker** [comment]
I'm with you.  I'll try to see if I can recompile without graphviz but I think the last time I tried, lldb complained about the missing dependency.  You can try it also if you like but the compilation takes a long time (like several hours).  I guess lldb is the new windows. ;)

**Comment by blu** [comment]
Ok, here are lldb's dependencies as of llvm 7.0.0 on my machine:
```
$ ldd `which lldb`
        libLLVMSupport.so.7 => /usr/local/lib/libLLVMSupport.so.7 (0xef662000)
        libpthread.so.0 => /usr/local/lib/libpthread.so.0 (0xef63f000)
        liblldb.so.7 => /usr/local/lib/liblldb.so.7 (0xeeade000)
        libstdc++.so.6 => /usr/local/lib/libstdc++.so.6 (0xee983000)
        libm.so.6 => /usr/local/lib/libm.so.6 (0xee90b000)
        libgcc_s.so.1 => /usr/local/lib/libgcc_s.so.1 (0xee8db000)
        libc.so.6 => /usr/local/lib/libc.so.6 (0xee7ee000)
        libz.so.1 => /usr/local/lib/libz.so.1 (0xee7c1000)
        librt.so.1 => /usr/local/lib/librt.so.1 (0xee7ab000)
        libdl.so.2 => /usr/local/lib/libdl.so.2 (0xee798000)
        libncurses.so.5 => /lib/libncurses.so.5 (0xee767000)
        libLLVMDemangle.so.7 => /usr/local/lib/libLLVMDemangle.so.7 (0xee746000)
        /lib/ld-linux-armhf.so.3 (0xef789000)
        libLLVMARMCodeGen.so.7 => /usr/local/lib/libLLVMARMCodeGen.so.7 (0xee532000)
        libLLVMARMAsmParser.so.7 => /usr/local/lib/libLLVMARMAsmParser.so.7 (0xee4b5000)
        libLLVMARMAsmPrinter.so.7 => /usr/local/lib/libLLVMARMAsmPrinter.so.7 (0xee496000)
        libLLVMARMDesc.so.7 => /usr/local/lib/libLLVMARMDesc.so.7 (0xee3cd000)
        libLLVMARMDisassembler.so.7 => /usr/local/lib/libLLVMARMDisassembler.so.7 (0xee389000)
        libLLVMARMInfo.so.7 => /usr/local/lib/libLLVMARMInfo.so.7 (0xee386000)
        libLLVMARMUtils.so.7 => /usr/local/lib/libLLVMARMUtils.so.7 (0xee381000)
        libLLVMIRReader.so.7 => /usr/local/lib/libLLVMIRReader.so.7 (0xee379000)
        libLLVMExecutionEngine.so.7 => /usr/local/lib/libLLVMExecutionEngine.so.7 (0xee35a000)
        libLLVMCore.so.7 => /usr/local/lib/libLLVMCore.so.7 (0xee100000)
        libLLVMObject.so.7 => /usr/local/lib/libLLVMObject.so.7 (0xee037000)
        libLLVMRuntimeDyld.so.7 => /usr/local/lib/libLLVMRuntimeDyld.so.7 (0xedfdf000)
        libLLVMTarget.so.7 => /usr/local/lib/libLLVMTarget.so.7 (0xedfd2000)
        libclangFrontend.so.7 => /usr/local/lib/libclangFrontend.so.7 (0xede9a000)
        libclangCodeGen.so.7 => /usr/local/lib/libclangCodeGen.so.7 (0xedad6000)
        libclangDriver.so.7 => /usr/local/lib/libclangDriver.so.7 (0xed96f000)
        libclangEdit.so.7 => /usr/local/lib/libclangEdit.so.7 (0xed95f000)
        libclangParse.so.7 => /usr/local/lib/libclangParse.so.7 (0xed88c000)
        libclangRewrite.so.7 => /usr/local/lib/libclangRewrite.so.7 (0xed879000)
        libclangRewriteFrontend.so.7 => /usr/local/lib/libclangRewriteFrontend.so.7 (0xed80f000)
        libclangSema.so.7 => /usr/local/lib/libclangSema.so.7 (0xed182000)
        libclangSerialization.so.7 => /usr/local/lib/libclangSerialization.so.7 (0xed01a000)
        libLLVMipo.so.7 => /usr/local/lib/libLLVMipo.so.7 (0xecf19000)
        libLLVMMCJIT.so.7 => /usr/local/lib/libLLVMMCJIT.so.7 (0xecf0c000)
        libclangBasic.so.7 => /usr/local/lib/libclangBasic.so.7 (0xecce3000)
        libLLVMDebugInfoDWARF.so.7 => /usr/local/lib/libLLVMDebugInfoDWARF.so.7 (0xecc6f000)
        libclangAST.so.7 => /usr/local/lib/libclangAST.so.7 (0xec94d000)
        libclangLex.so.7 => /usr/local/lib/libclangLex.so.7 (0xec896000)
        libLLVMDebugInfoPDB.so.7 => /usr/local/lib/libLLVMDebugInfoPDB.so.7 (0xec80d000)
        libLLVMBinaryFormat.so.7 => /usr/local/lib/libLLVMBinaryFormat.so.7 (0xec7fa000)
        libedit.so.0 => /usr/local/lib/libedit.so.0 (0xec7c0000)
        libpython2.7.so.1.0 => /usr/local/lib/libpython2.7.so.1.0 (0xec5df000)
        libxml2.so.2 => /usr/local/lib/libxml2.so.2 (0xec498000)
        libform.so.5 => /usr/lib/libform.so.5 (0xec48c000)
        libpanel.so.6 => /usr/local/lib/libpanel.so.6 (0xec479000)
        libLLVMMC.so.7 => /usr/local/lib/libLLVMMC.so.7 (0xec3cd000)
        libLLVMMCDisassembler.so.7 => /usr/local/lib/libLLVMMCDisassembler.so.7 (0xec3c5000)
        libLLVMAnalysis.so.7 => /usr/local/lib/libLLVMAnalysis.so.7 (0xec0dc000)
        libLLVMAsmPrinter.so.7 => /usr/local/lib/libLLVMAsmPrinter.so.7 (0xec037000)
        libLLVMCodeGen.so.7 => /usr/local/lib/libLLVMCodeGen.so.7 (0xebc2b000)
        libLLVMGlobalISel.so.7 => /usr/local/lib/libLLVMGlobalISel.so.7 (0xebbce000)
        libLLVMScalarOpts.so.7 => /usr/local/lib/libLLVMScalarOpts.so.7 (0xeb944000)
        libLLVMSelectionDAG.so.7 => /usr/local/lib/libLLVMSelectionDAG.so.7 (0xeb715000)
        libLLVMTransformUtils.so.7 => /usr/local/lib/libLLVMTransformUtils.so.7 (0xeb587000)
        libLLVMMCParser.so.7 => /usr/local/lib/libLLVMMCParser.so.7 (0xeb544000)
        libLLVMAsmParser.so.7 => /usr/local/lib/libLLVMAsmParser.so.7 (0xeb4e2000)
        libLLVMBitReader.so.7 => /usr/local/lib/libLLVMBitReader.so.7 (0xeb492000)
        libLLVMOption.so.7 => /usr/local/lib/libLLVMOption.so.7 (0xeb482000)
        libLLVMProfileData.so.7 => /usr/local/lib/libLLVMProfileData.so.7 (0xeb430000)
        libclangAnalysis.so.7 => /usr/local/lib/libclangAnalysis.so.7 (0xeb390000)
        libLLVMBitWriter.so.7 => /usr/local/lib/libLLVMBitWriter.so.7 (0xeb359000)
        libLLVMCoroutines.so.7 => /usr/local/lib/libLLVMCoroutines.so.7 (0xeb338000)
        libLLVMCoverage.so.7 => /usr/local/lib/libLLVMCoverage.so.7 (0xeb312000)
        libLLVMAggressiveInstCombine.so.7 => /usr/local/lib/libLLVMAggressiveInstCombine.so.7 (0xeb304000)
        libLLVMInstCombine.so.7 => /usr/local/lib/libLLVMInstCombine.so.7 (0xeb232000)
        libLLVMInstrumentation.so.7 => /usr/local/lib/libLLVMInstrumentation.so.7 (0xeb172000)
        libLLVMLTO.so.7 => /usr/local/lib/libLLVMLTO.so.7 (0xeb12b000)
        libLLVMLinker.so.7 => /usr/local/lib/libLLVMLinker.so.7 (0xeb10e000)
        libLLVMObjCARCOpts.so.7 => /usr/local/lib/libLLVMObjCARCOpts.so.7 (0xeb0ea000)
        libLLVMPasses.so.7 => /usr/local/lib/libLLVMPasses.so.7 (0xeb050000)
        libLLVMVectorize.so.7 => /usr/local/lib/libLLVMVectorize.so.7 (0xeafa6000)
        libLLVMDebugInfoCodeView.so.7 => /usr/local/lib/libLLVMDebugInfoCodeView.so.7 (0xeaf11000)
        libLLVMDebugInfoMSF.so.7 => /usr/local/lib/libLLVMDebugInfoMSF.so.7 (0xeaeff000)
        libncurses.so.6 => /usr/local/lib/libncurses.so.6 (0xeaeb2000)
        libutil.so.1 => /usr/local/lib/libutil.so.1 (0xeae9f000)
        libicui18n.so.62 => /usr/local/lib/libicui18n.so.62 (0xeac4a000)
        libicuuc.so.62 => /usr/local/lib/libicuuc.so.62 (0xeaac3000)
        libicudata.so.62 => /usr/local/lib/libicudata.so.62 (0xe9118000)
(END)
```
Points of interest:
* both `/usr/lib/libncurses.so.5` and `/usr/local/lib/libncurses.so.6` are used
* lldb itself does not directly depend on graphviz. But from https://lldb.llvm.org/build.html:

`If you wish to build the optional reference documentation, additional dependencies are required:

Graphviz (for the 'dot' tool).
doxygen (only if you wish to build the C++ API reference)
epydoc (only if you wish to build the Python API reference)`

So graphviz seems needed for building the ref docs, but is not needed for the functioning of lldb itself. Am I reading this right?

**Comment by JL2210** [comment]
Solution: `depends on 'graphviz' :=> build`
